game-core/src/main/java/games/strategy/triplea/delegate/RocketsFireHelper.java

```diff
@@ -35,7 +38,62 @@
 /**
  * Logic to fire rockets.
  */
-public class RocketsFireHelper {
+public class RocketsFireHelper implements Serializable {
+  private static final long serialVersionUID = -3694482580987118444L;
+  private final Set<Territory> attackingFromTerritories = new HashSet<>();
+  private final Map<Territory, Territory> attackedTerritories = new LinkedHashMap<>();
+  private final Map<Territory, Unit> attackedUnits = new LinkedHashMap<>();
+  private final GameData data;
```

Comment:
If you can find a way to obtain this `GameData` reference at runtime (e.g. from some other method parameter) and not have to serialize it, that would be great.  Back references to `GameData` in serializable types create circularities that have made it incredibly difficult to migrate away from Java Serialization as our default persistence format.

Reply:
Not 100% sure but I guess we can try it and see if any issues are found. Let's just hope they aren't found after going live.